### ISSUE TITLE
Completion always expands to full configured space_for_menu size.

### ISSUE THREAD

**Issue by hardkrash** [ipython/ipython#9786]
I have some objects that are quite full of properties.  So, I wanted to show more of the properties on completions.

This lead me to setting c.TerminalInteractiveShell.space_for_menu to 16

Works great for the big list object, but fails for a smaller object as it still push the prompt up 16 spaces.

I would prefer that this values would be a maximum for the height after the multicolumn split.

```min((math.ceil(num_completions/num_col), c.TerminalInteractiveShell.space_for_menu)```

In other words, if I expand out to three columns, then grow the list from there up to this limit, not the current behavior of grow out to this limit and then do the columns.

Adding the ability to take a percentage of the screen might be even nicer.

**Comment by takluyver** [comment]
Pinging @jonathanslenders - I think this would require new options in prompt_toolkit - though let me know if there's something already there that I don't know about!

**Comment by jonathanslenders** [comment]
Hi. This is not in prompt-toolkit yet. It would require ``FloatContainer.preferred_height`` [1] to look at height of both the content and the floating objects (the menu, in our case).
Maybe, I'll implement several expansion strategies, because I'm not sure whether everyone prefers exactly the same. (Currently, it has the advantage that the prompt is pushed up only once. If you go on, and press "Tab" for a second completion, it should stay in place.)

[1] https://github.com/jonathanslenders/python-prompt-toolkit/blob/master/prompt_toolkit/layout/containers.py#L389
